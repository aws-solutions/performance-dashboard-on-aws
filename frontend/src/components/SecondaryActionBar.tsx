/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React, { ReactNode } from "react";
import { useWindowSize } from "../hooks";

interface Props {
    children: ReactNode;
    className?: string;
    //Distance from the top of the page (in pixels) that this element will stick at
    stickyPosition?: number;
}

function SecondaryActionBar(props: Props) {
    const windowSize = useWindowSize();
    const isMobile = windowSize.width <= 600;

    return (
        <div
            className={`border-base-lighter border-1px shadow-2 z-300 radius-md padding-x-2 padding-top-2 padding-bottom-1 margin-top-2 margin-bottom-1 bg-white ${
                isMobile ? "" : "position-sticky "
            }${props.className}`}
            style={{ top: props.stickyPosition }}
        >
            {props.children}
        </div>
    );
}

export default SecondaryActionBar;
