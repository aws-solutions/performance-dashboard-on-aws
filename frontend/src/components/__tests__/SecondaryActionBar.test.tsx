/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { render } from "@testing-library/react";
import SecondaryActionBar from "../SecondaryActionBar";

test("renders a secondary action bar", async () => {
    const wrapper = render(
        <SecondaryActionBar stickyPosition={75}>
            <p>Test</p>
        </SecondaryActionBar>,
    );
    expect(wrapper.container).toMatchSnapshot();
});
