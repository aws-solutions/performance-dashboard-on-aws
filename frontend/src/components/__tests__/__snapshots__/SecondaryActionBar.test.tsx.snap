// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a secondary action bar 1`] = `
<div>
  <div
    class="border-base-lighter border-1px shadow-2 z-300 radius-md padding-x-2 padding-top-2 padding-bottom-1 margin-top-2 margin-bottom-1 bg-white position-sticky undefined"
    style="top: 75px;"
  >
    <p>
      Test
    </p>
  </div>
</div>
`;
